src/plugins/guided_onboarding/README.md

```diff
@@ -1,9 +1,75 @@
-# guidedOnboarding
+# Guided Onboarding
 
-A Kibana plugin
+This plugin contains the code for the Guided Onboarding project. The guided onboarding consists of guides for Solutions (Enterprise Search, Observability, Security) that can be completed as a checklist of steps. The guides help users to ingest their data and to navigate to the correct Solutions pages. 
+
+The guided onboarding plugin includes a client-side code for the UI and the server side code for the internal API. The server-side code is not intended for external use. 
+
+The client-side code registers a button in the Kibana header that controls the guided onboarding panel (checklist) depending on the current state. There is also an API service exposed from the client-side start contract. The API service is intended for external use by other plugins.
 
 ---
 
 ## Development
 
-See the [kibana contributing guide](https://github.com/elastic/kibana/blob/main/CONTRIBUTING.md) for instructions setting up your development environment.
+1. To enable the UI, add `guidedOnboarding.ui: true` to the file `KIBANA_FOLDER/config/kibana.dev.yml`. 
+
+2. Start Kibana with examples `yarn start --run-examples` to be able to see the guidedOnboardingExample plugin. 
+
+3. Navigate to `/app/guidedOnboardingExample` to start a guide and check the button in the header.
+
+## API service
+*Also see `KIBANA_FOLDER/examples/guided_onboarding_example` for code examples.*
+
+The guided onboarding plugin exposes an API service from its start contract that is intended to be used by other plugins. The API service allows consumers to access the current state of the guided onboarding process and manipulate it. 
+
+To use the API service in your plugin, declare the guided onboarding plugin as a dependency in the file `kibana.json` of your plugin. Add the API service to your plugin's start dependencies to rely on the provided TypeScript interface:
+```
+export interface AppPluginStartDependencies {
+  guidedOnboarding: GuidedOnboardingPluginStart;
+}
+```
+The API service is now available to your plugin in the setup lifecycle function of your plugin
+```
+// startDependencies is of type AppPluginStartDependencies
+const [coreStart, startDependencies] = await core.getStartServices();
+```
+or in the start lifecycle function of your plugin.
+```
+public start(core: CoreStart, startDependencies: AppPluginStartDependencies) {
+  ...
+}
+```
+
+### isGuideStepActive$(guideID: string, stepID: string): Observable\<boolean\>
+*Also see `KIBANA_FOLDER/examples/guided_onboarding_example/public/components/step_one.tsx`.*
+
+The API service exposes an Observable that contains a boolean value for the state of a specific guide step. For example, if your plugin needs to check if the "Add data" step of the Security guide is currently active, you could use the following code snippet. 
```

Comment:
I think it might be helpful if you add some information about the guide configs, i.e., where to find them and how to determine which step ID to use.

Reply:
On another note, we should probably go through and update the configs - at least the IDs and number of steps - now that we know the full flow for each solution. I can open up a quick PR to handle that.